packages/panna-sdk/src/react/components/collectibles/collectibles-list.tsx

```diff
@@ -142,12 +150,45 @@ export function CollectiblesList({ className }: CollectiblesListProps) {
                 </div>
               </AccordionTrigger>
               <AccordionContent className="grid grid-cols-2 gap-4">
+                {/* Since ERC-1155 allows owning multiple tokens with the same ID,
+                check value to display multiple tokens */}
                 {item.instances.map((instance, instanceIndex) => (
-                  <Card key={instanceIndex} className="p-0">
-                    <CardContent className="p-0">
-                      <CollectibleImageRenderer instance={instance} />
-                    </CardContent>
-                  </Card>
+                  <Fragment key={`${instance.id}-${instanceIndex}`}>
+                    {instance?.value ? (
+                      <>
+                        {Array.from({ length: Number(instance.value) }).map(
+                          (_, valueIndex) => (
+                            <Card
+                              key={`${instance.id}-${valueIndex}`}
+                              className="p-0"
+                            >
+                              <CardContent className="p-0">
+                                <CollectibleImageRenderer
+                                  instance={instance}
+                                  token={item.token}
+                                  setActiveView={setActiveView}
+                                  setActiveCollectible={setActiveCollectible}
+                                  setActiveToken={setActiveToken}
+                                />
+                              </CardContent>
+                            </Card>
+                          )
+                        )}
+                      </>
+                    ) : (
+                      <Card key={instanceIndex} className="p-0">
+                        <CardContent className="p-0">
+                          <CollectibleImageRenderer
+                            instance={instance}
+                            token={item.token}
+                            setActiveView={setActiveView}
+                            setActiveCollectible={setActiveCollectible}
+                            setActiveToken={setActiveToken}
+                          />
+                        </CardContent>
+                      </Card>
```

Comment:
To avoid code duplication and make maintenance easy, can we please create a new function to return the `Card` based on the passed arguments instead?

Reply:
Done.